jdbc-v2/src/test/java/com/clickhouse/jdbc/DataTypeTests.java

```diff
@@ -846,4 +855,64 @@ public void testDynamicTypesSimpleStatement() throws SQLException {
             }
         }
     }
+
+
+    @Test(groups = { "integration" })
+    public void testTypeConversions() throws Exception {
+        try (Connection conn = getConnection()) {
+            try (Statement stmt = conn.createStatement()) {
+                try (ResultSet rs = stmt.executeQuery("SELECT 1, 'true', '1.0', " +
+                        "toDate('2024-12-01'), toDateTime('2024-12-01 12:34:56'), toDateTime64('2024-12-01 12:34:56.789', 3), toDateTime64('2024-12-01 12:34:56.789789', 6), toDateTime64('2024-12-01 12:34:56.789789789', 9)")) {
+                    assertTrue(rs.next());
+                    assertEquals(rs.getInt(1), 1);
+                    assertEquals(String.valueOf(rs.getObject(1)), "1");
+                    assertEquals(rs.getObject(1, Integer.class), 1);
+                    assertEquals(rs.getObject(1, Long.class), 1L);
+                    assertEquals(String.valueOf(rs.getObject(1, new HashMap<String, Class<?>>(){{put(JDBCType.INTEGER.getName(), Integer.class);}})), "1");
+
+                    assertTrue(rs.getBoolean(2));
+                    assertEquals(String.valueOf(rs.getObject(2)), "true");
+                    assertEquals(rs.getObject(2, Boolean.class), true);
+                    assertEquals(String.valueOf(rs.getObject(2, new HashMap<String, Class<?>>(){{put(JDBCType.BOOLEAN.getName(), Boolean.class);}})), "true");
+
+                    assertEquals(rs.getFloat(3), 1.0f);
+                    assertEquals(String.valueOf(rs.getObject(3)), "1.0");
+                    assertEquals(rs.getObject(3, Float.class), 1.0f);
+                    assertEquals(rs.getObject(3, Double.class), 1.0);
+                    assertEquals(String.valueOf(rs.getObject(3, new HashMap<String, Class<?>>(){{put(JDBCType.FLOAT.getName(), Float.class);}})), "1.0");
+
+                    assertEquals(rs.getDate(4), Date.valueOf("2024-12-01"));
+                    assertTrue(rs.getObject(4) instanceof ZonedDateTime);//ZonedDateTime
```

Comment:
This is not true - according to spec we need to return sql.Date or sql.Timestamp